starport/templates/app/stargate/app/app.go.plush

```diff
@@ -250,6 +250,7 @@ func New(
 	// grant capabilities for the ibc and ibc-transfer modules
 	scopedIBCKeeper := app.CapabilityKeeper.ScopeToModule(ibchost.ModuleName)
 	scopedTransferKeeper := app.CapabilityKeeper.ScopeToModule(ibctransfertypes.ModuleName)
+	// this line is used by starport scaffolding # stargate/app/scopedKeeper
```

Comment:
This break compatibility for previously scaffolded app when you want to import wasm

If introducing new placeholders is the only way to go to be able to support importing wasm we should return an error to the user with the import command and tell them to downgrade starport to import wasm or eventually print the instruction to support it (which may be not possible with the old version of Cosmos SDK)

Reply:
I don't think we can keep backwards compatibility for changes like this nor create migration instructions for all different apps we scaffold in the past. Even if we attempt to it, it's pretty complex and hard to maintain. This is yet another unfortunate cons of placeholders.

Scaffolding related comamnds in-fact should fail they cannot find related placeholders in the source code, which might also happen if users manually delete them but this is not in the scope of this PR.

Reply:
> Scaffolding related commands in-fact should fail when they cannot find related placeholders in the source code, which might also happen if users manually delete them but this is not in the scope of this PR.

Yes this is definitely a refactor we must implement for the scaffolder.
Though I actually think that having a proper error if wasm cannot be imported because of backward incompatibility should be part of this PR. Because in this case this is not because users removed themself a placeholder (which we can't control anyway) but here this is part of a normal workflow: the user has scaffolded a blockchain in past, he now wants to have `wasm` in their app, the command will actually succeed but then serving will throw errors.

We could just verify if the placeholder exists or not in `app.go`, this is what we do when trying to create an IBC module with old scaffolded app that doesn't support this feature

Reply:
Otherwise I think we should have the scaffolder throwing errors before the next release

Reply:
If we start introducing placeholder spesific checks, this will get out of control pretty soon. Placeholders are already bad, these checks will make things even more complicated. I'd rather to fail execution if a certain placeholder cannot be found no matter if it's removed by the user or recently added by a new version of Starport.

Once we add the generic placeholder check, we can print the name of placeholder we're looking for to the console and print a help page in Starport docs so, users can see what's this about and take an action. We should solve this problem in docs rather than in code.

Reply:
https://github.com/tendermint/starport/issues/1136

Reply:
This is fine for me if w're able do this before `0.16.0` 👍

Placeholders are not convenient when we need to update things but I think we should absolutely avoid users being able to scaffold an app that can't be served after having followed a normal workflow